count_cells.py

```diff
@@ -153,8 +132,18 @@ def count_cells(self, matrix_file: Path) -> Union[int, None]:
 
     def count_cells_from_barcodes(self, barcodes_file: Path) -> Union[int, None]:
         with open_maybe_gz(barcodes_file,  'rt') as f:
-            next(f)  # skip header line
-            return sum(1 for _ in f)
+            line_count = 0
+            similarity_ratios = []  # first line compared against following few lines
+            first_line = next(f)  # grab first line to see if is a header or not
```

Comment:
I realize some thought went into it but unfortunately, if we have matrix companion files that lack a header we need to fix them rather which will render this heuristic unnecessary, or in the wrong place at best. If you can convince me that the heuristic is reliable, we should incorporate it into the convert_matrices utility instead.

For the time being, please remove this, and create another issue to investigate the missing header lines further.